test/test_find.cpp

```diff
@@ -80,14 +82,14 @@ TEST(test_find, find) {
 }
 
 TEST(test_find, findn) {
-  size_t ret0 = test_findn("", '/', 0, 0);
-  LOG((size_t)0, ret0);
+  size_t ret0 = test_findn("", '/', 0, SIZE_MAX);
+  LOG((size_t)SIZE_MAX, ret0);
```

Comment:
the size_t casts are unnecessary here and below now yeah?

Reply:
Yeah, for the SIZE_MAX checks, that's probably right.  I'll make that fix and submit a CI just to be sure.

Reply:
I wasn't aware of this but judging from [the warnings on OS X](http://ci.ros2.org/job/ci_osx/2929/warnings10Result/new/) `SIZE_MAX` may not fit into `size_t`.. this probably means we'll need to reconsider the return type of the find functions if we want to take this approach

Reply:
> I wasn't aware of this but judging from the warnings on OS X SIZE_MAX may not fit into size_t.. this probably means we'll need to reconsider the return type of the find functions if we want to take this approach

Arg, that is annoying.  I'll do a little more investigating to see what I can find here.

Reply:
I see what is going on.  The warning that clang is throwing is very misleading.  On x86_64 macOS, `size_t` is a long unsigned int, which is 64-bit unsigned.  SIZE_MAX is an unsigned long long, which is *also* 64-bit unsigned.  The problem is that the `LOG` statement is using `%zu` to print out the values, and an `unsigned long long` is not technically a `size_t`, even though they are compatible.  The solution is to make sure we still cast SIZE_MAX to a `size_t` before passing it to `LOG` (like what was there before).  I'll fix that, and put a comment in there for the next hapless soul who tries to remove it.